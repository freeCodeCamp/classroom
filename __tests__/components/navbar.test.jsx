import Navbar from '../../components/navbar';
import React from 'react';
import { SessionProvider } from 'next-auth/react';
import renderer from 'react-test-renderer';

describe('Navbar rendering correctly', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <SessionProvider session={{ user: { name: 'test user' } }}>
          <Navbar />
        </SessionProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
