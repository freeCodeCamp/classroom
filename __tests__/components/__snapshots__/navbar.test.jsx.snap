// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar rendering correctly renders correctly 1`] = `
<div
  className="h-[38px]"
>
  <div
    className="h-[38px] bg-fcc-gray-90 text-white flex items-center flex-wrap p-1"
  >
    <div
      className="hidden lg:flex block flex-1 justify-end"
    />
    <a
      className="flex items-center"
      href="/classes"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <span
        style={
          {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "inline-block",
            "height": 24,
            "margin": 0,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": 210,
          }
        }
      >
        <img
          alt="FreeCodecamp Logo"
          className=""
          data-nimg="fixed"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimages%2Ffcc_primary_large.png&w=640&q=75"
          srcSet="/_next/image?url=%2Fimages%2Ffcc_primary_large.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Ffcc_primary_large.png&w=640&q=75 2x"
          style={
            {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </span>
    </a>
    <div
      className="flex-1 inline-flex justify-end"
    >
      <div
        className="pl-2"
      >
        <button
          className="hover:bg-[#ffbf00] shadow-lg border-solid border-color: inherit; border-[1px] pl-4 pr-4 bg-fcc-primary-yellow text-black"
          onClick={[Function]}
        >
          Sign out
        </button>
      </div>
    </div>
  </div>
</div>
`;
